fuse_models/include/fuse_models/transform_sensor.hpp

```diff
@@ -0,0 +1,156 @@
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_MODELS__TRANSFORM_SENSOR_HPP_
+#define FUSE_MODELS__TRANSFORM_SENSOR_HPP_
+
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <memory>
+#include <string>
+
+#include <fuse_models/parameters/transform_sensor_params.hpp>
+#include <fuse_core/throttled_callback.hpp>
+
+#include <fuse_core/async_sensor_model.hpp>
+#include <fuse_core/uuid.hpp>
+
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <geometry_msgs/msg/twist_with_covariance_stamped.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+#include "tf2_msgs/msg/tf_message.hpp"
+
+namespace fuse_models
+{
+
+/**
+ * @brief An adapter-type sensor that produces pose constraints from AprilTag detections
+ *
+ * This sensor subscribes to a MessageType topic and creates orientation and pose variables and constraints.
+ *
+ * This sensor really just separates out the orientation, angular velocity, and linear acceleration
+ * components of the message, and processes them just like the Pose3D, Twist3D, and Acceleration3D
+ * classes.
+ *
+ * Parameters:
+ *  - device_id (uuid string, default: 00000000-0000-0000-0000-000000000000) The device/robot ID to
+ *                                                                           publish
+ *  - device_name (string) Used to generate the device/robot ID if the device_id is not provided
+ *  - queue_size (int, default: 10) The subscriber queue size for the pose messages
+ *  - topic (string) The topic to which to subscribe for the pose messages
+ *
+ * Subscribes:
+ *  - \p topic (MessageType) IMU data at a given timestep
+ */
+class TransformSensor : public fuse_core::AsyncSensorModel
+{
+public:
+  FUSE_SMART_PTR_DEFINITIONS(TransformSensor)
+  using ParameterType = parameters::TransformSensorParams;
+  using MessageType = tf2_msgs::msg::TFMessage;
+
+  /**
+   * @brief Default constructor
+   */
+  TransformSensor();
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~TransformSensor() = default;
```

Comment:
Is there a reason why the destructor is virtual? This sorta implies that it is okay to derive from this class?

Reply:
Yes, I think it is. I'm not 100% sure, but this is common across every fuse sensor in the repo.